tests/test_dse.py

```diff
@@ -487,10 +487,15 @@ def test_full_shape(self, rotate):
                                               'cire-rotate': rotate}))
 
         # Check code generation
+        trees = retrieve_iteration_tree(op1)
+        assert len(trees) == 3
+        assert all(i.dim.is_Incr for i in trees[1][1:4])
```

Comment:
this is not in master, because these are DSE tests, not DLE tests, so unless strictly necessary due to some special reasons, we don't check the integrity of blocking.

And yes, it does harm, because you're increasing complexity, and any _unnecessary_ increase in complexity does hard.

Reply:
The main issue is that indirectly, bf was checking that blocking exists..so we also need sth to ensure blocking exists at the correct place, though I admit that here is is indirectly checked as well... will reiterate..